### PR TITLE
 permission view

### PR DESCRIPTION
**Pull request by sakul-budhathoki**
Signed-off-by: Sakul <sakulbudhathoki977@gmail.com>

### CODE REVIEW

File: js/packages/components/main/Permissions.tsx [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+		}
+		if (navigateNext) {
+			props.navigation.navigate(navigateNext, {})
+		} else {
+			props.navigation.goBack()
+		}
+	}
+
+	const handleAppStateChange = async (nextAppState: string) => {
+		if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
```
Comment (n0izn0iz) [inline js/packages/components/main/Permissions.tsx:53]: could you add a comment with the rationale for checking permissions on AppState change pls? can't figure it out

**Review by n0izn0iz — APPROVED** [review]
LGTM

File: js/packages/components/main/Permissions.tsx (hunk 1 above)
Comment (sakul-budhathoki) [reply js/packages/components/main/Permissions.tsx:53]: Ok. If the permission is denied, then we will show user "Go to Setting" button to let them enable permission from Settings. After enabling it from settings, user will be back to our app.At that moment, we will check if permission is enabled or not using AppState change; if enabled we will navigate back. If not user will be on same screen. I hope it's clear now.

**Comment by zxxma** [comment]
Front (iOS) : ✅  LGTM

**Pull request merged**